File: 1295. Find Numbers with Even Number of Digits/Solution.java
public class Solution {
    // Given an array nums of integers,
    // return how many of them contain an even number of digits.
    

    int count = 0;

    public int findNumbers(int[] nums) {

        for (int i : nums) {
            if (isEven(i)) {
                System.out.println(isEven(i));
                count++;
            }
        }

        return count;
    }

    private boolean isEven(int i) {
        int num = digit(i);
        return num % 2 == 0;
    }

    public int digit(int num) {
        int count = 0;
        if (num == 0) {
            return 1;
        }

        while (num > 0) {
            count++;
            num = num / 10;

        }
        System.out.println(count);
        return count;
    }

    public static void main(String[] args) {
        Solution s = new Solution();
        int[] nums = { 12, 345, 23, 0, 7896 }; // Answer should be 2
        int ans = s.findNumbers(nums);
        System.out.println(ans);
    }

}

File: 1295. Find Numbers with Even Number of Digits/Solution.js
// /**
//  * @param {number[]} nums
//  * @return {number}
//  */

const findNumbers = function(nums) {
  let count = 0;
  for (let num of nums) {
    if (even(num)) {
      count++;
    }
  }
  console.log(count);
  return count;
};

const sohel = [123,2222,34,50,47]
console.log(findNumbers(sohel));

function even(num) {
  const variable = digit(num);
  return variable % 2 === 0;
}

function digit(num) {
  let count = 0;
  if (num === 0) {
    return 1;
  }
  while (num > 0) {
    count++;
    num = Math.floor(num / 10);
  }
  return count;
}
